### PR TITLE
feat: Add custom jump diagnostic headers

### PR DESCRIPTION
**Pull request by cooper-anderson**
This PR adds the ability to override the text that shows at the top of a jump diagnostic. I decided to make this since I felt the diagnostic sign was a bit too close to the left side of the window. Instead of just adding a margin, I decided to make the whole header text (before the other parts get added) configurable.

Todo:
- [x] Add configs for diagnostic headers
- [x] Make the config defaults match the existing header
- [x] Allow user to override headers in their settings
- [x] Add configs to the list of configurable settings in the README

### CODE REVIEW

File: lua/lspsaga/diagnostic.lua [hunk 1]
```diff
@@ -300,4 +300,13 @@ function M.lsp_diagnostic_sign(opts)
   end
 end
 
+function M.lsp_diagnostic_header(opts)
```
Comment (glepnir) [inline lua/lspsaga/diagnostic.lua:303]: why there use a function just like 
```lua
local config = require('lspsaga').config_values
local severity_icon = {
  config.error_header,
....
}
```

File: lua/lspsaga/init.lua [hunk 2]
```diff
@@ -38,6 +43,7 @@ function saga.init_lsp_saga(opts)
   local syntax = require 'lspsaga.syntax'
 
   handlers.overwrite_default(saga.config_values)
+  diagnostic.lsp_diagnostic_header(saga.config_values)
```
Comment (glepnir) [inline lua/lspsaga/init.lua:46]: same here. not need this.

**Review by glepnir — COMMENTED** [review]
no need usa overwrite funciton

**Comment by cooper-anderson** [comment]
Ah, didn't see that

:+1:

File: lua/lspsaga/diagnostic.lua [hunk 3]
```diff
@@ -10,10 +10,10 @@ local M = {}
 -- lsp severity icon
 -- 1:Error 2:Warning 3:Information 4:Hint
 local severity_icon = {
-  "  Error",
-  "  Warn",
-  "  Infor",
-  "  Hint"
+  config.config_values.error_header,
```
Comment (glepnir) [inline lua/lspsaga/diagnostic.lua:13]: statement is too long just use this
```lua
local config = require('lspsaga').config_values
config.error_header
```

**Review by glepnir — COMMENTED** [review]
short statement

**Comment by glepnir** [comment]
Hi can you resolve the conflicts with my new commits.

**Comment by cooper-anderson** [comment]
Conflicts resolved :+1:
Let me know if there's anything else that needs to get done.

**Pull request merged**